### PR TITLE
Add public APIs documentation

### LINKED ISSUE #6: Document public APIs
image

### PR DESCRIPTION
**Pull request by HugoHeneault**
Closes #6

I don't know why is the stream used [here](https://github.com/OwnWeb/flutter_map_tappable_polyline/pull/9/files#diff-d801a7e0cec4753b8106c998c8313a1bR76) is for?

### CODE REVIEW

File: lib/flutter_map_tappable_polyline.dart [hunk 1]
```diff
@@ -16,22 +16,33 @@ class TappablePolylineMapPlugin extends MapPlugin {
...
+  /// The ability to render only polylines in current view bounds
+  final bool polylineCulling;
+
   TappablePolylineLayerOptions(
       {this.polylines = const [],
       rebuild,
       this.onTap,
       this.pointerDistanceTolerance = 15,
-      polylineCulling = false})
-      : super(rebuild: rebuild, polylineCulling: polylineCulling);
```
Comment (tuarrep) [inline lib/flutter_map_tappable_polyline.dart:31]: Why this removal?

File: lib/flutter_map_tappable_polyline.dart [hunk 2]
```diff
@@ -56,8 +67,13 @@ class TaggedPolyline extends Polyline {
 }
 
 class TappablePolylineLayer extends StatelessWidget {
+  /// The options allowing tappable polyline tweaks
   final TappablePolylineLayerOptions polylineOpts;
+
+  /// The flutter_map [MapState]
   final MapState map;
+
+  // what is that stream for?
```
Comment (tuarrep) [inline lib/flutter_map_tappable_polyline.dart:76]: Can't remember, I'll check

Comment (tuarrep) [reply lib/flutter_map_tappable_polyline.dart:76]: Stream used by `flutter_map` to notify us a redraw is required

See `StreamBuilder` in `_build` method.

File: lib/flutter_map_tappable_polyline.dart (hunk 1 above)
Comment (HugoHeneault) [reply lib/flutter_map_tappable_polyline.dart:31]: This isn't a removal. `polylineCulling` is now a final property of `TappablePolylineLayerOptions`

File: lib/flutter_map_tappable_polyline.dart (hunk 2 above)
Comment (HugoHeneault) [reply lib/flutter_map_tappable_polyline.dart]: Done.

File: lib/flutter_map_tappable_polyline.dart (hunk 1 above)
Comment (tuarrep) [reply lib/flutter_map_tappable_polyline.dart:31]: You have to pass it to super

Comment (HugoHeneault) [reply lib/flutter_map_tappable_polyline.dart:31]: Changes done

**Pull request merged**